.github/workflows/ci-foundry.yml

```diff
@@ -0,0 +1,52 @@
+name: Foundry CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "contracts/**"
+      - "lib/**"
+      - "test-foundry/**"
+      - "*.lock"
+      - "Makefile"
+      - "remappings.txt"
+      - ".github/workflows/ci-foundry.yml"
```

Comment:
It does not seem useful, and it can be kind of dangerous if you do a PR outside of the scope that break test